validate/src/embed.rs

```diff
@@ -289,17 +287,38 @@ pub fn embed(embed: &Embed) -> Result<(), EmbedValidationError> {
                 kind: EmbedValidationErrorType::TitleTooLarge { chars },
             });
         }
-
-        total += chars;
     }
 
-    if total > EMBED_TOTAL_LENGTH {
-        return Err(EmbedValidationError {
-            kind: EmbedValidationErrorType::EmbedTooLarge { chars: total },
-        });
+    Ok(())
+}
+
+/// Get the total character count of an embed
+#[must_use]
+pub fn length(embed: &Embed) -> usize {
+    let mut length = 0;
+
+    length += embed
+        .author
+        .as_ref()
+        .map_or("", |author| author.name.as_str())
+        .len();
```

Comment:
I think this would be a cleaner way to write this:

```suggestion
    if let Some(author) = &embed.author {
        length += author.name.len();
    }
```

and we can apply this to the other option checks.